iothub/device/src/Transport/HttpClientHelper.cs

```diff
@@ -87,26 +87,31 @@ IWebProxy proxy
 
             _httpClientObj = handler != null ? new HttpClient(handler) : new HttpClient();
 #else
+
             if (httpClientHandler == null)
             {
                 httpClientHandler = new HttpClientHandler();
             }
-            httpClientHandler.SslProtocols = TlsVersions.Instance.Preferred;
-            httpClientHandler.CheckCertificateRevocationList = TlsVersions.Instance.CertificateRevocationCheck;
 
-            if (clientCert != null)
+            using (httpClientHandler)
             {
-                httpClientHandler.ClientCertificates.Add(clientCert);
-                _usingX509ClientCert = true;
-            }
+                httpClientHandler.SslProtocols = TlsVersions.Instance.Preferred;
+                httpClientHandler.CheckCertificateRevocationList = TlsVersions.Instance.CertificateRevocationCheck;
 
-            if (proxy != DefaultWebProxySettings.Instance)
-            {
-                httpClientHandler.UseProxy = proxy != null;
-                httpClientHandler.Proxy = proxy;
-            }
+                if (clientCert != null)
+                {
+                    httpClientHandler.ClientCertificates.Add(clientCert);
+                    _usingX509ClientCert = true;
+                }
+
+                if (proxy != DefaultWebProxySettings.Instance)
+                {
+                    httpClientHandler.UseProxy = proxy != null;
+                    httpClientHandler.Proxy = proxy;
+                }
 
-            _httpClientObj = new HttpClient(httpClientHandler);
+                _httpClientObj = new HttpClient(httpClientHandler);
```

Comment:
now you've saved this to the class but disposed the handler that you passed to it. That's not good. I think you need to save the handler to the class as a field as well, and dispose it with the `HttpClient` object